src/main/java/genesis/Genesis.java

```diff
@@ -0,0 +1,530 @@
+        }
+        for (int i = 0; i < tasks.size(); i++) {
+            System.out.println((i + 1) + ". " + tasks.get(i));
+        }
+    }
+    /**
+     * Marks the task with the given index.
+     *
+     * @param index Index of the task to mark.
+     */
+    public void markTask(int index) {
+        if (isValidIndex(index)) {
+            tasks.get(index).mark();
+            System.out.println("Nice! I've marked this task as done:\n" + tasks.get(index).description);
+        }
+    }
+    /**
+     * Unmarks the task with the given index.
+     *
+     * @param index Index of the task to unmark.
+     */
+    public void unmarkTask(int index) {
+        if (isValidIndex(index)) {
+            tasks.get(index).unmark();
+            System.out.println("Ok. I've marked this task as not done yet:\n" + tasks.get(index).description);
+        }
+    }
+    /**
+     * Deletes the task with the given index.
+     *
+     * @param index Index of the task to delete.
+     */
+    public void deleteTask(int index) {
+        if (isValidIndex(index)) {
+            Task removedTask = tasks.remove(index);
+            System.out.println("Noted. I have removed the following task: \n" + removedTask.toString() +
+                    "\nNow you have " + tasks.size() + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Checks if index given is valid.
+     *
+     * @param index Index of the task to check.
+     */
+
+    private boolean isValidIndex(int index) {
+        if (index < 0 || index >= tasks.size()) {
+            System.out.println("No such task exists!");
+            return false;
+        }
+        return true;
+    }
+}
+/**
+ * Handles loading tasks from and writing tasks to a file.
+ */
+class Storage {
+    protected TaskManager taskManager;
+    protected CommandParser commandParser;
+
+    /**
+     * Constructs a Storage object with the specified TaskManager.
+     *
+     * @param taskManager The TaskManager associated with this storage.
+     */
+
+    public Storage(TaskManager taskManager) {
+        this.taskManager = taskManager;
+    }
+
+    /**
+     * Loads tasks from a file and populates the TaskManager.
+     *
+     * @param commandParser The command parser used to handle commands.
+     * @param ui The UI component to handle user inputs.
+     */
+    public void loadTasks(CommandParser commandParser, Ui ui) {
+        File f = new File("data/tasks.txt");
+        if (f.exists()) {
+            try (Scanner s = new Scanner(f)) {
+                while (s.hasNext()) {
+                    ui.handleInput(s.nextLine(), true);
+                }
+
+            } catch (IOException e) {
+                System.out.println("Error loading tasks from file: " + e.getMessage());
+            }
+        } else {
+            System.out.println("Error. Data file does not exist!");
+        }
+    }
+
+    /**
+     * Writes the current list of tasks to a file.
+     */
+    public void writeTasks() {
```

Comment:
I like how simple your` writeTasks` function is, I think it is intuitive and easy to understand 👍🏻